src/m365/yammer/commands/yammer-search.spec.ts

```diff
@@ -0,0 +1,622 @@
+        "groups":  2,
+        "topics":  5,
+        "users":  4
+      },
+      "messages": {
+        "messages": []
+      },
+      "groups": [],
+      "topics": [],
+      "users": []        
+    };
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+    (command as any).items = [];
+  });
+
+  afterEach(() => {
+    Utils.restore([
+      request.get
+    ]);
+  });
+
+  after(() => {
+    Utils.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.YAMMER_SEARCH), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('correctly handles error', (done) => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      return Promise.reject({
+        "error": {
+          "base": "An error has occurred."
+        }
+      });
+    });
+
+    command.action(logger, { options: { debug: false } } as any, (err?: any) => {
+      try {
+        assert.strictEqual(JSON.stringify(err), JSON.stringify(new CommandError("An error has occurred.")));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('does not pass validation without parameters', () => {
+    const actual = command.validate({ options: {} });
+    assert.notStrictEqual(actual, false);
+  });
+
+  it('passes validation with one parameter', () => {
+    const actual = command.validate({ options: { queryText: '123123' } });
+    assert.notStrictEqual(actual, false);
+  });
+
+  it('passes validation with parameters', () => {
+    const actual = command.validate({ options: { queryText: '123', limit: 10, output: 'json' } });
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation with parameters', () => {
+    const actual = command.validate({ options: { queryText: '123', show: "summary", output: 'json' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails if a wrong option is passed', () => {
+    const actual = command.validate({ options: { queryText: '123', show: 'wrongOption' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes if a correct option is passed', () => {
+    const options = ['summary', 'messages', 'users', 'topics', 'groups'];
+    options.forEach((option) => {
+      const actual = command.validate({ options: { queryText: '123', show: option } });
+      assert.strictEqual(actual, true);
```

Comment:
When defining multiple asserts in a single test, always specify third parameter (message), that will help us understand which assert failed.